libsolidity/Compiler.cpp

```diff
@@ -100,7 +100,7 @@ void Compiler::appendConstructorCall(FunctionDefinition const& _constructor)
 	{
 		m_context << u256(argumentSize);
 		m_context.appendProgramSize();
-		m_context << u256(1); // copy it to byte one as expected for ABI calls
+		m_context << u256(g_functionIdentifierOffset); // copy it to byte four as expected for ABI calls
```

Comment:
True ... then perhaps rename it to `g_dataStartOffset` since, this is when the data starts. And if we make any other additions the semantics should not change.